mne/channels/channels.py

```diff
@@ -1007,11 +1011,134 @@ def _ch_neighbor_connectivity(ch_names, neighbors):
     ch_connectivity = np.eye(len(ch_names), dtype=bool)
     for ii, neigbs in enumerate(neighbors):
         ch_connectivity[ii, [ch_names.index(i) for i in neigbs]] = True
-
     ch_connectivity = sparse.csr_matrix(ch_connectivity)
     return ch_connectivity
 
 
+def find_ch_connectivity(info, ch_type):
+    """Find the connectivity matrix for the given data.
+
+    This function tries to infer the appropriate connectivity matrix template
+    for the given data. If a template is not found, the connectivity matrix is
+    computed using Delaunay triangulation based on 2d sensor locations.
+
+    Parameters
+    ----------
+    info : instance of mne.measuerment_info.Info
+        The measurement info.
+    ch_type : str
+        The channel type for computing the connectivity matrix. Currently
+        supports 'mag', 'grad' and 'eeg'.
+
+    Returns
+    -------
+    ch_connectivity : scipy.sparse matrix
+        The connectivity matrix.
+    ch_names : list
+        The list of channel names present in connectivity matrix.
+
+    See Also
+    --------
+    read_ch_connectivity
+
+    Notes
+    -----
+    .. versionadded:: 0.15
+    """
+    (has_vv_mag, has_vv_grad, is_old_vv, has_4D_mag, ctf_other_types,
+     has_CTF_grad, n_kit_grads, has_any_meg, has_eeg_coils,
+     has_eeg_coils_and_meg, has_eeg_coils_only) = _get_ch_info(info)
```

Comment:
once montage is set there is no leftover information about it, right (like montage name)? I understand it would be too difficult to try to guess cap type just from channel positions, so I'm ok with this, just asking. :)